R/settings_utils.R

```diff
@@ -68,24 +68,24 @@ find_config <- function(filename) {
     dirname(filename)
   }
 
-  ## check for a file in the current directory
-  linter_config <- file.path(path, linter_file)
-  if (isTRUE(file.exists(linter_config))) {
-    return(linter_config)
-  }
-
-  ## next check for a file higher directories
-  linter_config <- find_config2(path)
-  if (isTRUE(file.exists(linter_config))) {
-    return(linter_config)
-  }
-
-  ## next check for a file in the user directory
-  # cf: rstudio@bc9b6a5 SessionRSConnect.R#L32
-  home_dir <- Sys.getenv("HOME", unset = "~")
-  linter_config <- file.path(home_dir, linter_file)
-  if (isTRUE(file.exists(linter_config))) {
-    return(linter_config)
+  # List possible .lintr file locations
+  file_locations <- c(
+    # Check for a file in the current directory
+    file.path(path, linter_file),
+    # Next check for a file in the .github/linters directory
+    file.path(path, ".github/linters", linter_file),
+    # Next check for a file in higher directories
+    find_config2(path),
```

Comment:
`find_config2()` should also look for `.github/linters` (i.e. as a sibling directory to `getwd()`)
This will ensure e.g. code running in `tests/testthat` from within a package will use a package-level `.github/linters/.lintr`

Reply:
Hi @AshesITR ,

Sorry I'm not sure what you mean here. is it to combine lines 76 & 78 into one using `find_config2(file.path(path, ".github", "linters"))`
Or something different as you think there is a bug?

Reply:
I'm thinking that the code of `find_config2()`, which currently looks for a `.lintr` file in a parent directory, should be modified to also look for a `.github/linters/.lintr` file. This would make it so `lint("pkg/tests/testthat/test-my_stuff.R")` could find a config at `pkg/.github/linters/.lintr` as well.

Reply:
@AshesITR Were you thinking of something along these lines?

```diff
--- a/R/settings_utils.R
+++ b/R/settings_utils.R
@@ -90,7 +90,10 @@ find_config2 <- function(path) {
   config <- basename(getOption("lintr.linter_file"))
   path <- normalizePath(path, mustWork = FALSE)

-  while (!has_config(path, config)) {
+  while (
+    !has_config(path, config)
+    && !has_config(file.path(path, ".github", "linters"), config)
+  ) {
     path <- dirname(path)
     if (is_root(path)) {
       return(character())

```

Making this change would mean that we could remove the corresponding line from the `file_locations` definition in `find_config` (we can also remove the “current directory” logic, which is also covered by `find_config2` if I am not mistaken!).

This avoids having to hard-code the path `.github/linters` in two separate locations. — In fact, it would mean that `find_config` would not need to be changed at all (although that change removes redundancy and is probably a good idea regardless).

Reply:
Yes, something like that looks about right.

Reply:
Alright, I have a candidate implementation which fixes the unit test and implements this.

The question is: what is the easiest/cleanest way of adding it to this PR? Or should I create a new PR? Either way it is based off @tonyk7440’s initial changes, so these would be incorporated when merging a new PR instead of this one.

Reply:
@tonyk7440 can you give edit permissions to @klmr on your fork (or at least on this branch)?

Reply:
Hi @klmr, I think you should have access to edit the fork now, let me know if you don't

Reply:
I do, thanks! I will  push the changes as soon as I find time outside of work hours, which should happen in the next 48 hours.